### PR TITLE
Integrate ESLint in www/

### LINKED ISSUE #7963: Integrate eslint on www/
## Summary

We should run eslint on www/ as part of the Travis build, and additionally we should fix any issues that eslint identifies.

(Also we could run this as a git commit hook, but I'm less opinionated on that!)

### Motivation

Linting is generally valuable at catching some helpful code quality errors, and introducing linting as a git hook or as part of Travis will catch those errors early before something happens in a later phase (e.g. build phase).

In general, getting ESLint passing without any failures signifies that for at least code quality things that can be automated, we have a quality product :)

cc @berraknil if you're curious!

### PR DESCRIPTION
**Pull request by yeskunall**
Integrated ESLint into www/
- [x] Add `.eslintrc.json` (this was already there)
- [x] Add script to lint code
- [x] Fix ESlint lint errors

```shell
node/gatsby/www on  feat-add-eslint-www [$!] is 📦  v2.0.0-beta via ⬢ v10.3.0
❯ npm run lint:js

> gatsbyjs.org@2.0.0-beta lint:js /Users/kimchi/Documents/dev/node/gatsby/www
> eslint --ext .js,.jsx src/

Warning: React version not specified in eslint-plugin-react settings. See https://github.com/yannickcr/eslint-plugin-react#configuration.

/Users/kimchi/Documents/dev/node/gatsby/www/src/components/chart.js
  13:12  error  Strings must use backtick                        quotes
  13:23  error  Strings must use backtick                        quotes
  13:34  error  Strings must use backtick                        quotes
  13:45  error  Strings must use backtick                        quotes
  13:56  error  Strings must use backtick                        quotes
  17:19  error  Strings must use backtick                        quotes
  22:14  error  Strings must use backtick                        quotes
  24:12  error  Strings must use backtick                        quotes
  27:12  error  Strings must use backtick                        quotes
  28:20  error  Strings must use backtick                        quotes
  31:22  error  Strings must use backtick                        quotes
  39:20  error  Strings must use backtick                        quotes
  40:16  error  Strings must use backtick                        quotes
  41:25  error  Strings must use backtick                        quotes
  42:16  error  Strings must use backtick                        quotes
  46:20  error  Strings must use backtick                        quotes
  47:16  error  Strings must use backtick                        quotes
  48:25  error  Strings must use backtick                        quotes
  49:16  error  Strings must use backtick                        quotes
  64:13  error  Strings must use backtick                        quotes
  65:17  error  Strings must use backtick                        quotes
  69:9   error  Strings must use backtick                        quotes
  75:13  error  Strings must use backtick                        quotes
  77:15  error  Strings must use backtick                        quotes
  88:39  error  Expected block statement surrounding arrow body  arrow-body-style

/Users/kimchi/Documents/dev/node/gatsby/www/src/components/evaluation-table.js
   20:9   warning  Missing "key" prop for element in array  react/jsx-key
  117:17  warning  Missing "key" prop for element in array  react/jsx-key
  118:17  warning  Missing "key" prop for element in array  react/jsx-key
  123:23  warning  Missing "key" prop for element in array  react/jsx-key
  127:23  warning  Missing "key" prop for element in array  react/jsx-key
  164:23  warning  Missing "key" prop for element in array  react/jsx-key

/Users/kimchi/Documents/dev/node/gatsby/www/src/components/function-list.js
   99:43  error    Strings must use backtick                   quotes
  102:37  error    Strings must use backtick                   quotes
  126:17  warning  Missing "key" prop for element in iterator  react/jsx-key

/Users/kimchi/Documents/dev/node/gatsby/www/src/pages/features.js
  46:5  warning  Missing "key" prop for element in array  react/jsx-key
  49:5  warning  Missing "key" prop for element in array  react/jsx-key
  52:5  warning  Missing "key" prop for element in array  react/jsx-key
  55:5  warning  Missing "key" prop for element in array  react/jsx-key
  58:5  warning  Missing "key" prop for element in array  react/jsx-key
  64:5  warning  Missing "key" prop for element in array  react/jsx-key
  67:5  warning  Missing "key" prop for element in array  react/jsx-key
  68:5  warning  Missing "key" prop for element in array  react/jsx-key
  69:5  warning  Missing "key" prop for element in array  react/jsx-key
  70:5  warning  Missing "key" prop for element in array  react/jsx-key

/Users/kimchi/Documents/dev/node/gatsby/www/src/views/community/community-header.js
   9:7   error  'CommunityHeaderLink' is assigned a value but never used  no-unused-vars
  35:13  error  'forHire' is assigned a value but never used              no-unused-vars
  35:22  error  'hiring' is assigned a value but never used               no-unused-vars

/Users/kimchi/Documents/dev/node/gatsby/www/src/views/starter-library/starter-list.js
  15:66  error  Expected to return a value at the end of arrow function  consistent-return
  60:13  error  'gatsbyDependencies' is assigned a value but never used  no-unused-vars

✖ 49 problems (32 errors, 17 warnings)
  27 errors, 0 warnings potentially fixable with the `--fix` option.
```

I'll try my best to remove most of these errors. How does this look for a start? 😄

<hr />
Closes #7963.

### CODE REVIEW

**Comment by yeskunall** [comment]
All ESLint errors fixed! 👌🏽

File: www/src/views/community/community-header.js [hunk 1]
```diff
@@ -6,6 +6,7 @@ import Checkmark from "./check.svg"
 import Button from "../../components/button"
 import ArrowForwardIcon from "react-icons/lib/md/arrow-forward"
 
+// eslint-disable-next-line
```
Comment (Manoz) [inline www/src/views/community/community-header.js]: We can remove the hole function here or uncomment the creator links below

**Comment by yeskunall** [comment]
Thanks for the feedback @Manoz. I'll fix up the PR and ping you again if that's okay.

**Comment by pieh** [comment]
I'll close this because we've added global eslint config which takes care of `www` linting as well, so this is not needed anymore. There are few warnings on `www` and `examples` (mostly `react/jsx-key`), but no errors.

**Pull request closed without merging**